server/message_hooks.go

```diff
@@ -238,47 +239,32 @@ func (p *Plugin) SetChatReaction(teamsMessageID, srcUser, channelID, emojiName s
 	}
 
 	var teamsMessage *clientmodels.Message
-	tx, err := p.store.BeginTx()
+
+	mutex, err := cluster.NewMutex(p.API, "post_mutex_"+chatID+teamsMessageID)
```

Comment:
This seems slightly better than locking the Posts table, but slightly worse in that we're effectively writing out new rows to the KV store all the time.

When we pivot to driving this from shared channels, there will only be one instance processing these hooks, so the lock won't be necessary at all.

What are the risks if we simply don't lock at all here? I'm not certain of the "critical section" being guarded here.

Reply:
That was trying to fix a problem of reentrance of updates on reactions/post updates, that had a race condition and endup generating some inconsistencies. The point here is to avoid that.

I think you are right, the shared channels should solve this problem.

About the risk, if we don't lock at all there, probably it is going to bring back some problem with reactions, getting inconsistencies there, and even entering in some "looping" there updating a high number of times the reactions.

Reply:
I would pass this through QA, and see if everything goes as expected, if that is the case, merge this and wait until Shared Channels machinery is used, and remove the mutexes from there.

Reply:
> About the risk, if we don't lock at all there, probably it is going to bring back some problem with reactions, getting inconsistencies there, and even entering in some "looping" there updating a high number of times the reactions.

Can you elaborate a bit on this? I'd like to understand what went wrong before we had the transactional locks. It would be great to test this somehow too -- I'm guessing the point of contention was to have a whole bunch of emojis added to a post... but then what was it that "went wrong"? The loop is especially concerning, but I'm wondering how that would happen.

Reply:
that reintroduces the the bugs #327 and #330

Reply:
I mean, it reintroduces the bug when we create the transactions, and the mutex in theory should fix that

Reply:
The reality, I'm not 100% sure if we also need to apply the mutex at the MSTeams activity handler to avoid properly the race conditions, because "locking the whole table" was a very intrusive approach.

Reply:
Discussed offline: we agreed to preserve the semantics here for now and revisit more holistically once we're leveraging shared channels as the substrate.